delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/projects/ProjectsImpl.scala

```diff
@@ -0,0 +1,272 @@
+package ch.epfl.bluebrain.nexus.delta.service.projects
+
+import java.util.UUID
+
+import akka.actor.typed.ActorSystem
+import akka.persistence.query.Offset
+import cats.effect.Clock
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.{Acl, AclAddress, AclCollection}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectCommand.{CreateProject, DeprecateProject, UpdateProject}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRejection.{OwnerPermissionsFailed, RevisionNotFound, UnexpectedInitialState}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectState.Initial
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.{Pagination, SearchParams, SearchResults}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, Envelope}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.UUIDF
+import ch.epfl.bluebrain.nexus.delta.sdk.{Acls, Organizations, ProjectResource, Projects}
+import ch.epfl.bluebrain.nexus.delta.service.cache.{KeyValueStore, KeyValueStoreConfig}
+import ch.epfl.bluebrain.nexus.delta.service.projects.ProjectsImpl.{entityType, projectTag, ProjectsAggregate, ProjectsCache}
+import ch.epfl.bluebrain.nexus.delta.service.syntax._
+import ch.epfl.bluebrain.nexus.sourcing._
+import ch.epfl.bluebrain.nexus.sourcing.processor.EventSourceProcessor._
+import ch.epfl.bluebrain.nexus.sourcing.processor.ShardedAggregate
+import ch.epfl.bluebrain.nexus.sourcing.projections.StreamSupervisor
+import com.typesafe.scalalogging.Logger
+import monix.bio.{IO, Task, UIO}
+import monix.execution.Scheduler
+
+final class ProjectsImpl private (
+    agg: ProjectsAggregate,
+    eventLog: EventLog[Envelope[ProjectEvent]],
+    index: ProjectsCache,
+    acls: Acls,
+    ownerPermissions: Set[Permission],
+    serviceAccount: Identity.Subject
+)(implicit base: BaseUri)
+    extends Projects {
+
+  private val component: String = "projects"
+
+  override def create(ref: ProjectRef, fields: ProjectFields)(implicit
+      caller: Identity.Subject
+  ): IO[ProjectRejection, ProjectResource] =
+    (eval(
+      CreateProject(
+        ref,
+        fields.description,
+        fields.apiMappings,
+        fields.baseOrGenerated(ref),
+        fields.vocabOrGenerated(ref),
+        caller
+      )
+    ) <* applyOwnerPermissions(ref, caller)).named("createProject", component)
```

Comment:
the `.named(...)` applies only to `applyOwnerPermissions(..., ...)` function when using the left fish